docs: add missing Admin API Plugin endpoints (#9050)
apache/apisix: docs/en/latest/admin-api.md

| ------ | ----------------------------------- | ------------ | ---------------------------------------------- |
| GET    | /apisix/admin/plugins/list          | NULL         | Fetches a list of all Plugins.                 |
| GET    | /apisix/admin/plugins/{plugin_name} | NULL         | Fetches the specified Plugin by `plugin_name`. |

> | GET         | /apisix/admin/plugins?all=true      | NULL         | Get all properties of all plugins. |

Reviewer: Add deprecate note for `all=true`.

@Revolyssup btw, does `apisix/admin/plugins` do what `/apisix/admin/plugins?all=true` do after the change? Or is the subsystem URL param _required_ for the`/apisix/admin/plugins` endpoint

Reviewer 2: @kayx23 The only way to return all results is with `all=true`. Even with `subsystem`, plugin name has to be provided. The mechanism of getting all plugins at once was deprecated with the usage of all=true.

| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
| GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
| PUT    | /apisix/admin/plugins/reload       | NULL         | Reloads the plugin according to the changes made in code |

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Gets the `{PLUGIN}` schema if it is supported in Stream/L4 subsystem. |

Reviewer: ```suggestion
| GET    | apisix/admin/plugins/{plugin_name}?subsystem=stream | NULL | Gets properties of a specified plugin if it is supported in Stream/L4 subsystem. |
```

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=http   | NULL | Gets the `{PLUGIN}` schema if it is supported in HTTP/L7 subsystem. |

Reviewer: ```suggestion
| GET    | apisix/admin/plugins/{PLUGIN}?subsystem=http   | NULL | Gets properties of a specified plugin if it is supported in HTTP/L7 subsystem. |
```

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=http   | NULL | Gets properties of a specified plugin if it is supported in HTTP/L7 subsystem. |

Reviewer: ```suggestion
| GET    | apisix/admin/plugins/{plugin_name}?subsystem=http   | NULL | Gets properties of a specified plugin if it is supported in HTTP/L7 subsystem. |

:::caution

The interface of getting properties of all plugins via `/apisix/admin/plugins?all=true` will be deprecated soon.

:::
```

| ------ | ----------------------------------- | ------------ | ---------------------------------------------- |
| GET    | /apisix/admin/plugins/list          | NULL         | Fetches a list of all Plugins.                 |
| GET    | /apisix/admin/plugins/{plugin_name} | NULL         | Fetches the specified Plugin by `plugin_name`. |
| GET         | /apisix/admin/plugins?all=true      | NULL         | Get all properties of all plugins. |

> | GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Get properties of all Stream plugins.|

Reviewer: ```suggestion
| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
```
...and the one above as well for consistency

| GET         | /apisix/admin/plugins?all=true&subsystem=stream| NULL | Gets properties of all Stream plugins.|
| GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
| PUT    | /apisix/admin/plugins/reload       | NULL         | Reloads the plugin according to the changes made in code |

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Checks if the `{PLUGIN}` is supported in Stream/L4 mode. |

Reviewer: ```
| GET    | /apisix/admin/plugins?all=true&subsystem=http | NULL | Gets properties of all HTTP plugins. |
```

Note the wording here is `Gets properties of`. Please make sure we're consistent (change either one of those).

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Checks if the `{PLUGIN}` is supported in Stream/L4 mode. |

Author: > I think it should be `Fetches the specified stream Plugin by `plugin_name` ` `Fetches the specified HTTP Plugin by `plugin_name` `

why? 🤔 It didn't make sense to me.

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Checks if the `{PLUGIN}` is supported in Stream/L4 mode. |

Reviewer 2: @shreemaan-abhishek This endpoint not only checks if the plugin exists in that subsystem but returns the plugin if it does exist in that subsystem.

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Checks if the `{PLUGIN}` is supported in Stream/L4 mode. |

Author: So is this better?

`Gets the {PLUGIN} if it is supported in XYZ mode`

> | GET    | apisix/admin/plugins/{PLUGIN}?subsystem=stream | NULL | Checks if the `{PLUGIN}` is supported in Stream/L4 mode. |

Reviewer: > ... but returns the plugin if it does exist in that subsystem.

@shreemaan-abhishek you can do a test to see the output . If I recall correctly it returns the plugin **schema**.
